### ISSUE TITLE
Cannot update enum value using native query for postgresEnumType

### ISSUE THREAD

**Issue by RachanaHV** [vladmihalcea/hibernate-types#239]
Hi Sir,
I am facing similar issue when i am trying to update the enum value using native query. I have added this below EnumType
public class EnumConversion extends org.hibernate.type.EnumType {

private static final long serialVersionUID = 1L;
    public void nullSafeSet(
            PreparedStatement st,
            Object value,
            int index,
            SharedSessionContractImplementor session)
            throws HibernateException, SQLException {
        if (value == null) {
            st.setNull(index, Types.OTHER);
        } else {
            st.setObject(index, value.toString(), Types.OTHER);
        }
    }
and my entity class looks like this
@entity
@table(name = "target",schema = "public")
@typedef(name="enumUserType", typeClass=EnumConversion.class)
public class Target {

 @Enumerated(EnumType.STRING)
 @Column(name = "try_1_status",columnDefinition="try_1_status")
 @Type(type="enumUserType")
 private TryStatus try1Status;
}

and my native query is like this

@Modifying
@Transactional
@Query(value = "update target set try_1_status=:status ,update_time=:time, try_1_date=:time  where iccid=:iccid", nativeQuery = true)
public void updateTry1Status(@Param("time") Timestamp time, @Param("iccid") String iccid, @Param("status") TryStatus status);
I am getting an error below error
2020-May-23 11:20:53:099 AM ERROR SqlExceptionHelper - ERROR: column "try_1_status" is of type try_status but expression is of type character varying
Hint: You will need to rewrite or cast the expression.
Position: 32
2020-May-23 11:20:53:125 AM ERROR CampaignRunnerScheduler - Caught Exception

I have tried my best to find possible solutions still i am not able execute the query
I can insert the enum value to DB (Postgres) using conversion but not able to do update.
Please help me .

**Comment by vladmihalcea** [comment]
It's very hard to follow the description  of the issue because of the lack of formatting.

If you manage to provide a replicating test case in this repository, without Spring, I'll reopen the issue.

**Comment by RachanaHV** [comment]
Hi, 

I mean to say that I have imported the below dependency and I am using version 2.9.10
<dependency>
		<groupId>com.vladmihalcea</groupId>
		<artifactId>hibernate-types-52</artifactId>
	<version>${hibernate-types.version}</version>
</dependency>

I am able to save the enum value with ref to your article
https://vladmihalcea.com/the-best-way-to-map-an-enum-type-with-jpa-and-hibernate/

Now I want to update the enum value and I am using the native query to update it.
below is the my query:

@Modifying
@transactional
@query(value = "update target set **try_1_status=:status** ,update_time=:time, try_1_date=:time where iccid=:iccid", nativeQuery = true)
public void updateTry1Status(@param("time") Timestamp time, @param("iccid") String iccid, **@param("status") TryStatus status)**;

**How can I change my query so that the query accepts the enum variable as input?**

I am getting below error in postgres


2020-May-23 11:48:22:184 AM **ERROR SqlExceptionHelper - ERROR: column "try_1_status" is of type try_status but expression is of type bytea
  Hint: You will need to rewrite or cast the expression.
  Position: 32**
2020-May-23 11:48:22:208 AM ERROR CampaignRunnerScheduler - Caught Exception
org.springframework.dao.InvalidDataAccessResourceUsageException: could not execute statement; SQL [n/a]; nested exception is org.hibernate.exception.SQLGrammarException: could not execute statement
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.convertHibernateAccessException(HibernateJpaDialect.java:281) ~[spring-orm-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:255) ~[spring-orm-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.translateExceptionIfPossible(AbstractEntityManagerFactoryBean.java:528) ~[spring-orm-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.dao.support.ChainedPersistenceExceptionTranslator.translateExceptionIfPossible(ChainedPersistenceExceptionTranslator.java:61) ~[spring-tx-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.dao.support.DataAccessUtils.translateIfNecessary(DataAccessUtils.java:242) ~[spring-tx-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:153) ~[spring-tx-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.data.jpa.repository.support.CrudMethodMetadataPostProcessor$CrudMethodMetadataPopulatingMethodInterceptor.invoke(CrudMethodMetadataPostProcessor.java:149) ~[spring-data-jpa-2.2.6.RELEASE.jar:2.2.6.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:95) ~[spring-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:212) ~[spring-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at com.sun.proxy.$Proxy139.updateTry1Status(Unknown Source) ~[?:?]
	at com.ota.cm.scheduler.CampaignRunnerScheduler.reportCurrentTime(CampaignRunnerScheduler.java:169) [classes/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_144]
	at org.springframework.scheduling.support.ScheduledMethodRunnable.run(ScheduledMethodRunnable.java:84) [spring-context-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) [spring-context-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_144]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_144]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_144]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_144]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_144]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_144]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_144]

**Comment by vladmihalcea** [comment]
> How can I change my query so that the query accepts the enum variable as input?

Most likely, you need to pass the Type during the bind. That will not be possible via `@Query`, but via a `createNativeQuery` call. 

So, you have to use [this Hibernate setParameter](https://docs.jboss.org/hibernate/orm/5.3/javadocs/org/hibernate/query/Query.html#setParameter-java.lang.String-java.lang.Object-):

```
entityManager
.createNativeQuery(
	"update target set try_1_status=:status ,update_time=:time, try_1_date=:time where iccid=:iccid"	
)
.unwrap(NativeQuery.class)
.setParameter("status", status, PostgreSQLEnumType.INSTANCE)
```

And, `EnumConversion` is not needed. You need to use the `PostgreSQLEnumType` offered by this project.
